pipcompilemulti/dependency.py

```diff
@@ -102,7 +102,8 @@ def __init__(self, line):
         regular = self.RE_DEPENDENCY.match(line)
         if regular:
             self.package = regular.group('package')
-            self.version = regular.group('version').strip()
+            _version = regular.group('version').strip()
+            self.version = re.sub(r"(?<=\.)0+(?=\d+)", "", _version)  # strip leading zeros
```

Comment:
Looks good. Let's compile the regex once along with other patterns (and lose `+` after `\d`):
```
RE_IGNORED_ZEROS = re.compile(r"(?<=\.)0+(?=\d)")
self.version = self.RE_IGNORED_ZEROS.sub("", regular.group('version').strip())
```

Reply:
Done!